src/dmd/e2ir.d

```diff
@@ -5890,10 +5910,9 @@ private elem *appendDtors(IRState *irs, elem *er, size_t starti, size_t endi)
             {
                 *pe = el_combine(edtors, erx);
             }
-            else if ((tybasic(erx.Ety) == TYstruct || tybasic(erx.Ety) == TYarray) &&
-                     !(erx.ET && type_size(erx.ET) <= 16))
+            else if (elemIsLvalue(erx))
             {
-                /* Expensive to copy, to take a pointer to it instead
```

Comment:
Not retaining this behavior for expensive-to-copy things?

Reply:
It is retained actually, structs bigger than two registers match the given definition of lvalue, i.e they are always lvalues in the backend regardless of the frontend's judgement. As far the backend is concerned what lives in registers is an rvalue and  what lives in memory is an lvalue.